www/_posts/2017-11-20-migrate-from-cordova-globalization-plugin.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+author:
+    name: Vishal Mishra
+    url: https://twitter.com/tweetsbymishra
+title:  "Migrating from the Globalization Plugin"
+categories: news
+tags: cordova globalization
+---
+
+## Migrating from the Cordova Globalization Plugin
+
+The Cordova Globalization Plugin was created to obtain information and perform operations based on a user’s locale, language and timezone when most mobile platforms could not make a distinction between these settings. With the new API arrivals in the browser, we can now use the [ECMA Internationalization API](https://www.ecma-international.org/ecma-402/1.0/) for achieving this goal on iOS, Android,Windows devices and desktop browsers. Hence, this cordova plugin is not needed any more and will be sunset soon.
+
+### Migrating from the plugin to the Internationalization API
+
+The cordova globalization plugin defines a global navigator.globalization object which provides various methods to access a user’s locale, language and timezone. To get the preferred language from the browser, the navigator.globalization.getPreferredLanguage method was used as shown below:
+
+```js
+navigator.globalization.getPreferredLanguage(function (language) {          
+    console.log('language: ' + language.value + '\n');
+}, function () { 
+    console.log('Error getting language\n'); 
+});
+```
+
+The current locale could be found out using:
+
+```js
+navigator.globalization.getLocaleName(function (locale) {          
+    console.log('locale: ' + locale.value + '\n');
+}, function () {
+    console.log('Error getting locale\n'); 
+});
+```
+
+The [ECMA Internationalization API](https://www.ecma-international.org/ecma-402/1.0/) provides the `Intl` object which provides language sensitive string comparison, number formatting, and date and time formatting. 
+First we should check if the API is supported by the browser:
+
+```js
+if (window.Intl && typeof window.Intl === 'object') {
+    console.log('API available');
+}
+```
+
+The preferred language can be found out from the browser by using the `navigator` object:
+
+```js
+console.log(navigator.language);
+```
+
+The locale name can be found out using the `Intl.getCanonicalLocales(locales)` method as shown below:
```

Comment:
Where does `locales` come from here? Is this to be used on the result of `navigator.language`?